SwiftCompilerSources/Sources/Optimizer/Utilities/EscapeUtils.swift

```diff
@@ -420,6 +420,28 @@ fileprivate struct EscapeWalker<V: EscapeVisitor> : ValueDefUseWalker,
           return .continueWalk
         }
         return isEscaping
+
+      case .AtomicLoad, .AtomicStore, .AtomicRMW, .CmpXChg:
```

Comment:
I think this is not correct. Atomic loads and stores should be treated exactly like `LoadInst` and `StoreInst`, which is not the case here.

Reply:
Stores walk up to their destination address, but for atomic store our "destination address" is the builtin raw pointer value, but escape analysis would have already walked that value and should be walking down the uses of this pointer (which is how we get to this builtin). If we wanted to do what store does, we'd walk up to the pointer value which is redundant because we've already visited it.